plugins/helper.js

```diff
@@ -151,12 +151,19 @@ async function createBuildEvent(apiUri, token, buildEvent, retryStrategyFn) {
     return new Promise((resolve, reject) => {
         requestretry(formatOptions('POST', `${apiUri}/v4/events`, token, buildEvent, retryStrategyFn), (err, res) => {
             if (!err) {
+                logger.info(
+                    `POST /v4/events/${buildEvent.buildId} completed with attempts, ${res.statusCode}, ${res.attempts}`
+                );
                 if (res.statusCode === 201) {
                     return resolve(res);
                 }
-                if (res.statusCode !== 201) {
```

Comment:
is this removed because `requestretry` set `err` for error cases and we don't have to check for other statuscode ?

Reply:
@jithine We are now checking for 201 and 200 as /v4/events somehow api returns both. For other statuses we will just reject. Retry would trigger for non 2xx and not 404.